HabitLogger.BBualdo/DatabaseLibrary/DbContext.cs

```diff
@@ -0,0 +1,636 @@
+﻿using DatabaseLibrary.Models;
+using Microsoft.Data.Sqlite;
+using System.Globalization;
+
+namespace DatabaseLibrary;
+
+public class DbContext
+{
+  private SqliteConnection _Connection { get; set; }
+
+  public DbContext()
+  {
+    _Connection = new SqliteConnection("Data Source=habits.db");
+    CreateTables();
+    SeedHabitTable();
+    SeedRecordTable();
+  }
+
+  public bool CreateHabit()
+  {
+    Console.Clear();
+    Console.WriteLine("Enter a name of your habit (running, meditating, reading, swimming, etc.):\n");
+    string? name = Console.ReadLine();
+
+    while (string.IsNullOrEmpty(name))
+    {
+      Console.WriteLine("\nName can't be blank.\n");
+      name = Console.ReadLine();
+    }
+
+    Console.WriteLine("\nEnter a measurement unit for your habit (kilometers, times, pages, minutes, etc.):\n");
+    string? unit = Console.ReadLine();
+
+    while (string.IsNullOrEmpty(unit))
+    {
+      Console.WriteLine("\nUnit can't be blank.\n");
+      unit = Console.ReadLine();
+    }
+
+    using (_Connection)
+    {
+      _Connection.Open();
+
+      string checkIfExistsQuery = $"SELECT COUNT(*) FROM habit WHERE name='{name}'";
+
+      using (SqliteCommand checkCommand = new SqliteCommand(checkIfExistsQuery, _Connection))
+      {
+        int matchingHabits = Convert.ToInt32(checkCommand.ExecuteScalar());
+
+        if (matchingHabits > 0)
+        {
+          Console.WriteLine("\nThat habit already exists. Press any key to return to Main Menu.\n");
+          Console.ReadKey();
+          _Connection.Close();
```

Comment:
Because you're using the using statement on the connection, it's not needed to close it manually. [This example](https://learn.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlconnection.close?view=dotnet-plat-ext-8.0#examples) is for Sql Server, but it's the same principle.

Reply:
Okay, that is what I thought but in the tutorial, pablo did it manually so despite of thinking that GC should take care of it I followed that method. Furthermore, I red a lot of topics about how closing important is database connection closing, so had to be double sure 😁